lib/fsm/follow.go

```diff
@@ -65,60 +66,50 @@ func (e *PlanRolledBackEvent) isTerminalEvent() bool { return true }
 func FollowOperationPlan(ctx context.Context, getPlan GetPlanFunc) <-chan PlanEvent {
 	ch := make(chan PlanEvent, 100)
 	// Send an initial batch of events from the initial state of the plan.
-	plan, err := getPlan()
-	if err != nil {
-		logrus.WithError(err).Error("Failed to load plan.")
-	}
-	if plan != nil {
-		events := getPlanEvents(GetPlanProgress(*plan), *plan)
-		// recreate channel with enough room for all events + some room for additional events
-		ch = make(chan PlanEvent, len(events)+10)
```

Comment:
I removed the resizing logic in favor of moving this part into a separate function to reuse in the loop. It did not seem beneficial (e.g. no recorded evidence it was optimizing anything).

Reply:
@knisbet you added this in your scalability work (#2426). Can you double check?

Reply:
This wasn't an optimization, it was to fix that when we had more than 100 events the follow would only show the first 100 events of history and then only show new events. So on large plans it would print the 100 events, and then jump to like step 5000 if it was still running.

Reply:
See my comment above - I don't think the buffer is needed in the first place.